src/net/client/mod.rs

```diff
@@ -0,0 +1,60 @@
+use async_trait::async_trait;
+use hyper::client::HttpConnector;
+use hyper::{Body, Client, Uri};
+use hyper_tls::HttpsConnector;
+use thiserror::Error as ThisError;
+
+use super::Command;
+
+#[derive(ThisError, Debug)]
+pub enum MpcClientError {
+    #[error("invalid host address")]
+    InvalidHostAddress(String),
+
+    #[error("network connection error")]
+    NetworkConnection(#[from] hyper::Error),
+}
+
+#[async_trait]
+pub trait MpcHandle {
+    async fn execute(&self, command: Command) -> Result<Vec<u8>, MpcClientError>;
+}
+
+pub struct MpcHttpConnection {
+    client: Client<HttpsConnector<HttpConnector>>,
+    addr: Uri,
+}
+
+#[async_trait]
+impl MpcHandle for MpcHttpConnection {
+    async fn execute(&self, command: Command) -> Result<Vec<u8>, MpcClientError> {
+        match command {
+            Command::Echo(s) => self.echo(&s).await,
+        }
+    }
+}
+
+impl MpcHttpConnection {
+    #[must_use]
+    pub fn new(addr: &str) -> Self {
+        // this works for both http and https
```

Comment:
I am curious how you made it work with self-signed certificate, I had to explicitly disable certificate validation because hyper client just refused to establish connection, so I had to do this

```
let conn = TlsConnector::builder()
            .danger_accept_invalid_certs(true)
            .build()
            .unwrap();
```

Reply:
For this PR, I assumed I'd only connect to CA-signed sites. I'll add a self-signed cert connection in the next PR.